src/groups/mqb/mqbblp/mqbblp_queuehandle.cpp

```diff
@@ -483,32 +493,67 @@ void QueueHandle::deliverMessageImpl(
 
-    // Create an event to dispatch delivery of the message to the client
-    mqbi::DispatcherClient* client = d_clientContext_sp->client();
-    mqbi::DispatcherEvent*  event  = client->dispatcher()->getEvent(client);
-    (*event)
-        .setType(mqbi::DispatcherEventType::e_PUSH)
-        .setSource(d_queue_sp.get())
-        .setGuid(msgGUID)
-        .setQueueId(id())
-        .setMessagePropertiesInfo(d_queue_sp->schemaLearner().demultiplex(
-            d_schemaLearnerPushContext,
-            attributes.messagePropertiesInfo()))
-        .setSubQueueInfos(subQueueInfos)
-        .setMsgGroupId(msgGroupId)
-        .setCompressionAlgorithmType(attributes.compressionAlgorithmType());
-
-    if (message) {
-        event->setBlob(message);
+    if (d_clientContext_sp->inlineClient() == 0) {
+        // Create an event to dispatch delivery of the message to the client
+        mqbi::DispatcherClient* client = d_clientContext_sp->client();
+        mqbi::DispatcherEvent*  event = client->dispatcher()->getEvent(client);
+        (*event)
+            .setType(mqbi::DispatcherEventType::e_PUSH)
+            .setSource(d_queue_sp.get())
+            .setGuid(msgGUID)
+            .setQueueId(id())
+            .setMessagePropertiesInfo(d_queue_sp->schemaLearner().demultiplex(
+                d_schemaLearnerPushContext,
+                attributes.messagePropertiesInfo()))
+            .setSubQueueInfos(subscriptions)
+            .setMsgGroupId(msgGroupId)
+            .setCompressionAlgorithmType(
+                attributes.compressionAlgorithmType());
+
+        if (message) {
+            event->setBlob(message);
+        }
+
+        client->dispatcher()->dispatchEvent(event, client);
+        return;  // RETURN
     }
 
-    client->dispatcher()->dispatchEvent(event, client);
+    mqbi::InlineResult::Enum result =
+        d_clientContext_sp->inlineClient()->sendPush(
+            msgGUID,
+            id(),
+            message,
+            attributes,
+            d_queue_sp->schemaLearner().demultiplex(
+                d_schemaLearnerPushContext,
+                attributes.messagePropertiesInfo()),
+            subscriptions);
+
+    if (result == mqbi::InlineResult::e_SUCCESS) {
+        for (bmqp::Protocol::SubQueueInfosArray::size_type i = 0;
+             i < subscriptions.size();
+             ++i) {
+            const bsl::shared_ptr<Subscription>& subscription =
+                d_subscriptions[subscriptions[i].id()];
+
+            if (subscription->stats()) {
+                subscription->stats()->onEvent(
+                    mqbstat::QueueStatsClient::EventType::e_PUSH,
+                    message ? message->length() : 0);
+            }
+        }
+    }
+    else if (d_throttledDroppedPutMessages.requestPermission()) {
+        BALL_LOG_WARN << "Queue '" << d_queue_sp->description()
+                      << "' failed to PUSH " << msgGUID
+                      << " with error: " << result;
```

Comment:
This is logged only when we don't have an inline client, is it okay to skip that case for logging?